lib/puppet/transaction/additional_resource_generator.rb

```diff
@@ -22,11 +24,10 @@ def generate_additional_resources(resource)
     generated = [generated] unless generated.is_a?(Array)
     generated.collect do |res|
       @catalog.resource(res.ref) || res
-    end.each do |res|
-      priority = @prioritizer.generate_priority_contained_in(resource, res)
-      add_resource(res, resource, priority)
+    end.reverse_each do |res|
```

Comment:
Not sure I understand this change. I brief description of why we do this in reverse would be useful.